Metagrating3D/metagrating-meep.py

```diff
@@ -0,0 +1,151 @@
+                         center=mp.Vector3(0,0,-0.5*sz+dpml),
+                         component=src_cmpt)]
+
+    sim = mp.Simulation(resolution=resolution,
+                        cell_size=cell_size,
+                        sources=sources,
+                        default_material=SiO2,
+                        boundary_layers=boundary_layers,
+                        k_point=k_point)
+
+    flux = sim.add_mode_monitor(fcen,
+                                0,
+                                1,
+                                mp.ModeRegion(center=mp.Vector3(0,0,0.5*sz-dpml),
+                                          size=mp.Vector3(px,py,0)))
+
+    sim.run(until_after_sources=mp.stop_when_dft_decayed())
+
+    input_flux = mp.get_fluxes(flux)
+
+    sim.reset_meep()
+
+    weights = np.genfromtxt('device1.csv',delimiter=',')
+    Nx, Ny = weights.shape
+
+    geometry = [mp.Block(size=mp.Vector3(mp.inf,mp.inf,dpml+dsub),
+                         center=mp.Vector3(0,0,-0.5*sz+0.5*(dpml+dsub)),
+                         material=SiO2),
+                mp.Block(size=mp.Vector3(px,py,gh),
+                         center=mp.Vector3(0,0,-0.5*sz+dpml+dsub+0.5*gh),
+                         material=mp.MaterialGrid(grid_size=mp.Vector3(Nx,Ny,1),
+                                                  medium1=mp.air,
+                                                  medium2=Si,
+                                                  weights=weights))]
+
+    sim = mp.Simulation(resolution=resolution,
+                        cell_size=cell_size,
+                        sources=sources,
+                        geometry=geometry,
+                        boundary_layers=boundary_layers,
+                        k_point=k_point)
+
+    flux = sim.add_mode_monitor(fcen,
+                                0,
+                                1,
+                                mp.ModeRegion(center=mp.Vector3(0,0,0.5*sz-dpml),
+                                              size=mp.Vector3(px,py,0)))
+
+    sim.run(until_after_sources=mp.stop_when_dft_decayed())
+
+    res = sim.get_eigenmode_coefficients(flux,
+                                         mp.DiffractedPlanewave((1,0,0),
+                                                                mp.Vector3(0,0,1),
+                                                                0 if P_pol else 1,
+                                                                1 if P_pol else 0))
+
+    coeffs = res.alpha
+    Tmeep = abs(coeffs[0,0,0])**2 / input_flux[0]
+
+    n1 = 1.0
+    n2 = nSiO2
+    Tfresnel = 1 - (n1-n2)**2 / (n1+n2)**2
+    diff_eff_P = 0.9114 # P-pol. diffraction efficiency computed using Reticolo (RCWA)
+    diff_eff_S = 0.9514 # S-pol. diffraction efficiency computed using Reticolo (RCWA)
+    # convert diffraction efficiency into transmittance in the Z direction
+    Treticolo = (diff_eff_P if P_pol else diff_eff_S) * Tfresnel * math.cos(theta_d)
```

Comment:
This conversion should not be necessary, since apparently reticolo is computing the power efficiency: the Poynting flux per unit cell in the diffraction order divided by the Poynting flux per unit cell in the incident wave.